lib/waste_exemptions_engine.rb

```diff
@@ -60,6 +61,13 @@ def use_last_email_cache
       change_string_to_boolean_for(@use_last_email_cache)
     end
 
+    def address_host=(value)
+      @address_host = value
```

Comment:
Why do we do this with the address host but not the Companies House host?

Reply:
If we intend to keep the `attr_reader` then we really should be. I've chucked this together to try and illustrate

```ruby
class OtherConfig
  attr_accessor :host
end

class MainConfig
  attr_reader :host

  def initialize
    @other = OtherConfig.new
  end

  def host=(value)
    @other.host = value
  end
end

my_config = MainConfig.new
my_config.host = "example.com"
my_config.host
# nil

class FixedConfig
  attr_reader :host

  def initialize
    @other = OtherConfig.new
  end

  def host=(value)
    @host = value
    @other.host = value
  end
end

my_fixed_config = FixedConfig.new
my_fixed_config.host = "example.com"
my_fixed_config.host
# "example.com"
```

Essentially the `attr_reader :companies_house_host` is creating a hidden instance variable which we don't see and our custom setter isn't referencing when it comes to the companies house host.

So if you were to call it right now it would return `nil`. Putting the `@address_host = value` ensures a call to `foo.address_host` will return the value set.

But having figured this out with this PR, once merged we really should decide whether to bother with the attr_readers (so fix them) or delete them and just have the setters.

Ping me if this still doesn't make sense.

Reply:
Cool, makes sense to me :)